discovery/marathon/marathon_test.go

```diff
@@ -326,6 +328,32 @@ func TestMarathonZeroTaskPorts(t *testing.T) {
 	}
 }
 
+func Test500ErrorHttpResponseWithValidJSONBody(t *testing.T) {
+	var (
+		ch     = make(chan []*targetgroup.Group, 1)
+		client = fetchApps
+	)
+	// Simulate 500 error with valid JSON response
+	respHandler := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Header().Set("Content-Type", "application/json")
+		io.WriteString(w, `{}`)
+	}
+	// create test server with handler
```

Comment:
Comments should be full sentences, starting with a capital letter and ending with a full stop.

Reply:
Done. Please check.